ipxe-builder/buildipxe.sh

```diff
@@ -89,6 +89,7 @@ fi
 sed -i 's/^\/\/#define[ \t]CONSOLE_SERIAL/#define\tCONSOLE_SERIAL/g' \
     "config/console.h"
 
+# shellcheck disable=SC2068
```

Comment:
I don't think this is needed. It is a proper array so it should work to quote it

Reply:
it does not because of make as far as I understood, I have tested it

Reply:
It is not array at all, it is a string. Hence it cannot have quotes, but also it should not have `[@]` part.

Reply:
I will push a fix, root cause of the thing is that I have accidentally not used the array syntax, it has to be an array for make to work, string with 2 space separated value does not work for some reason.

Reply:
The one with the shellcheck ignore works but I get why you would not prefer it.

Reply:
> I will push a fix, root cause of the thing is that I have accidentally not used the array syntax, it has to be an array for make to work, string with 2 space separated value does not work if I handle it as a regular string for some reason.

I don't think double-space is a thing working with arrays. If you want to use arrays, you `declare -a foo=()` and then you add stuff to them `foo+=("more" "stuff" "goes here").  Just declaring variable an array and then assigning string to simply manipulates its element 0.

```console
$ declare -a foo=()
$ foo="bar  baz"
$ declare -p foo
declare -a foo=([0]="bar  baz")

$ foo="$foo  fizz  buzz"
$ declare -p foo
declare -a foo=([0]="bar  baz  fizz  buzz")
```

Reply:
And I mean it won't work with the "" even if this variable declared as a string and used as a string without the [@] either.

Reply:
If it would be array, quotes would be correctly like @lentzi90 said. It would expand correctly into multiple arguments. As a string, quoting it makes it a single argument, which is problematic here. For this PR, shellcheck ignore + no quotes is correct, but you also need to remove `[@]` since you're dealing with a string, and not an array.

In general, we (and everybody writing Bash :smiley: ) would be better off more arrays and quotes instead of DIY build-your-argument-string.

Reply:
Now just ignore `SC2086` instead of `SC2068` and its golden for now.